install.sh

```diff
@@ -3,123 +3,404 @@
+_SUBITEM="\t$_ITEM"
+_SUBIMPORTANT="\t$_IMPORTANT"
+_SUBQUESTION="\t$_QUESTION"
 
-#Error handling
-#Kill 0 to kill subshells as well
+
+# ERROR HANDLING
 __err() {
-	printf "\n[!] Installation $_FAILED!\n"
-	kill 0
+	printf "\n$_IMPORTANT Installation $_FAILED on line $1.\n"
+	exit 1
+}
+
+__int() {
+	printf "\n$_IMPORTANT Installation \e[91mCANCELLED\e[0m.\n"
+	exit 1
 }
-trap __err ERR INT
+
+trap '__err $LINENO' ERR
+trap '__int' INT
+
 set -o errexit
 set -o errtrace
 set -o pipefail
 
-# Fix $HOME for users under standard sudo
+
+# PERMISSIONS GADGET
+# The user must run the build process, but root must install
+# software. In order to make sure the appropriate users
+# take the right actions, we call sudo in the script itself.
+
+# Prevent user running sudo themselves
 if [ ! -z ${SUDO_USER+x} ]; then
-	HOME="$( getent passwd $SUDO_USER | cut -d: -f6 )"
+	printf "Please run the RITA installer without sudo.\n"
+	exit 1
+fi
+
+# Root is running the script without sudo
+if [ "$EUID" = "0" ]; then
+	_ELEVATE=""
+else
+	printf "$_IMPORTANT The RITA installer requires root privileges for some tasks. \n"
+	printf "$_IMPORTANT \"sudo\" will be used when necessary. \n"
+	_SUDO="$(type -fp sudo)"
+	if [ -z $_SUDO ]; then
+		printf "\"sudo\" was not found on the system. Please log in as root \n"
+		printf "before running the installer, or install \"sudo\". \n"
+		exit 1
+	fi
+	$_SUDO -v
+	if [ $? -ne 0 ]; then
+		printf "The installer was unable to elevate privileges using \"sudo\". \n"
+		printf "Please make sure your account has \"sudo\" privileges. \n"
+	fi
+	# _ELEVATE is separate from _SUDO since environment variables may need
+	# to be passed
+	_ELEVATE="$_SUDO"
 fi
 
-# Make sure to source the latest .bashrc
-# Hack the PS1 variable to get around ubuntu .bashrc
-_OLD_PS1=$PS1
-PS1=" "
-# Hack the interactive flag to get around other .bashrc's
-set -i
-# Make sure weirdness doesn't happen with autocomplete/ etc
-set -o posix
+# ENTRYPOINT
+__entry() {
+	_UNINSTALL_RITA=false
+
+	# Optional Dependencies
+	_INSTALL_BRO=true
+	_INSTALL_MONGO=true
 
-source $HOME/.bashrc
+	# Install locations
+	_INSTALL_PREFIX=/usr/local
+	_CONFIG_PATH=/etc/rita
+	_VAR_PATH=/var/lib/rita
 
-# Clean up our hacks
-set +o posix
-set +i
-PS1=$_OLD_PS1
-unset _OLD_PS1
+	# FOR an OPT style installation
+	# NOTE: RITA itself must be changed to agree with the
+	# _CONFIG_PATH and _VAR_PATH
+	# _INSTALL_PREFIX=/opt/rita
+	# _CONFIG_PATH=/etc/opt/rita
+	# _VAR_PATH=/var/opt/rita
 
-__help() {
-	__title
+	# Parse through command args
+	while [[ $# -gt 0 ]]; do
+		case $1 in
+			-h|--help)
+				# Display help and exit
+				__help
+				exit 0
+				;;
+			-u|--uninstall)
+				_UNINSTALL_RITA=true
+				_INSTALL_BRO=false
+				_INSTALL_MONGO=false
+				;;
+			--disable-bro)
+				_INSTALL_BRO=false
+				;;
+			--disable-mongo)
+				_INSTALL_MONGO=false
+				;;
+			--prefix)
+				shift
+				_INSTALL_PREFIX="$1"
+				;;
+			--version)
+				shift
+				_RITA_VERSION="$1"
```

Comment:
Add the ability to install a different branch, and remove the (unrecorded) issue where RITA will fail at runtime if built outside of git.